packages/chevrotain/src/parse/grammar/checks.ts

```diff
@@ -53,10 +37,13 @@ import {
   IGrammarValidatorErrorMessageProvider,
   IParserDefinitionError
 } from "./types"
-import dropRight from "lodash/dropRight"
-import compact from "lodash/compact"
 import { tokenStructuredMatcher } from "../../scan/tokens"
 
+// ES2019 Array.prototype.flatMap
+function flatMap<U, R>(arr: U[], callback: (x: U, idx: number) => R[]): R[] {
+  return Array.prototype.concat.apply([], arr.map(callback))
+}
```

Comment:
`flatMap` is called repeatedly in this file (`checks.ts`), and I found that inlining the above definition would've made the code less readable. So I opted to put a tiny non-exported helper function at the top of the file.

The same definition is also repeated in `llk_lookahead.ts`. I considered putting it into `parse/parser/utils`, but as it's just a one-liner, and it can [go away](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap) once we bump the minimum ES version to 2019, I figured it's easier to just define this one-liner in each file.